src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ONE_ELEMENT = 1;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        size = 0;
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity > 0) {
+            size = initialCapacity;
+            elementData = (T[]) new Object[initialCapacity];
+        } else if (initialCapacity == 0) {
+            elementData = (T[]) new Object[0];
+        } else {
+            throw new RuntimeException("Number of sheet elements is incorrect");
+        }
+    }
+
+    public void checkingIndex(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("Out of bounds search index");
+        }
+    }
+
+    public void ensureCapacity() {
```

Comment:
make them private and move to the end of class

Reply:
Done.